sample/src/main/java/com/canhub/cropper/sample/camera_java/app/SCameraFragmentJava.java

```diff
@@ -0,0 +1,338 @@
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import androidx.activity.result.ActivityResultLauncher;
+import androidx.activity.result.contract.ActivityResultContracts;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.core.content.FileProvider;
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentActivity;
+import androidx.viewpager.widget.ViewPager;
+
+import com.canhub.cropper.CropImage;
+import com.canhub.cropper.CropImageView;
+import com.canhub.cropper.sample.SCropResultActivity;
+import com.canhub.cropper.sample.camera.domain.CameraEnumDomain;
+import com.canhub.cropper.sample.camera.domain.SCameraContract;
+import com.canhub.cropper.sample.camera.presenter.SCameraPresenter;
+import com.example.croppersample.R;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.io.File;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Locale;
+
+import static android.graphics.Color.RED;
+import static android.graphics.Color.WHITE;
+
+public class SCameraFragmentJava extends Fragment implements SCameraContract.View {
+
+    static final int CODE_PHOTO_CAMERA = 811917;
+    static final String DATE_FORMAT = "yyyyMMdd_HHmmss";
+    static final String FILE_NAMING_PREFIX = "JPEG_";
+    static final String FILE_NAMING_SUFFIX = "_";
+    static final String FILE_FORMAT = ".jpg";
+    static final String AUTHORITY_SUFFIX = ".fileprovider";
+    static final int CUSTOM_REQUEST_CODE = 8119153;
+
+    private ViewHolder viewHolder;
+    private SCameraContract.Presenter presenter = new SCameraPresenter();
+    private Uri photoUri;
+    private ActivityResultLauncher<String> requestPermissionLauncher =
+            registerForActivityResult(new ActivityResultContracts.RequestPermission(), isGranted -> {
+                if (isGranted) {
+                    // Permission is granted. Continue the action or workflow in your
+                    // app.
+                } else {
+                    // Explain to the user that the feature is unavailable because the
+                    // features requires a permission that the user has denied. At the
+                    // same time, respect the user's decision. Don't link to system
+                    // settings in an effort to convince the user to change their
+                    // decision.
+                }
+            });
+
+    public static SCameraFragmentJava newInstance(){
+        return new SCameraFragmentJava();
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View rootView = inflater.inflate(R.layout.fragment_camera, container, false);
+        viewHolder = new ViewHolder(rootView);
+        rootView.setTag(viewHolder);
+
+        return rootView;
```

Comment:
Can we use viewBinding? more safer for null cases, so we can focus on the library components